Snowplow/SPUtilities.m

```diff
@@ -145,13 +145,21 @@ + (NSString *) getAppleIdfv {
 }
 
 + (NSString *) getCarrierName {
-    NSString * carrierName = nil;
 #if SNOWPLOW_TARGET_IOS
-    CTTelephonyNetworkInfo *netinfo = [[CTTelephonyNetworkInfo alloc] init];
-    CTCarrier *carrier = [netinfo subscriberCellularProvider];
-    carrierName = [carrier carrierName];
+    CTTelephonyNetworkInfo *networkInfo = [CTTelephonyNetworkInfo new];
+    if (@available(iOS 12.1, *)) {
+        // `serviceSubscribersCellularProviders` has a bug in the iOS 12.0 so we use it from iOS 12.1
+        NSDictionary<NSString *,CTCarrier *> *services = [networkInfo serviceSubscriberCellularProviders];
+        NSArray<CTCarrier *> *carriers = services.allValues;
+        // From iOS 12, iPhones with eSIMs can return multiple carrier providers.
+        // We can't prefer anyone of them so we track the first reported.
+        return [carriers.firstObject carrierName];
```

Comment:
Can this be empty if there is no eSim and no Sim Card?

Reply:
Yes it could be empty. But ObjC is much more relaxed than other languages about the null values.
`carriers.firstObject` is nil if the array is empty.
Also in ObjC world _calling a method_ is the same of _sending a message_. So it tolerates _sending a message to nil_. So, in this case, all the nil checking can be removed.

Reply:
Ah yes, I knew this but I think I still had my Java brain on!